modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -154,6 +154,78 @@ PERF_TEST_P_(BuildPyr_CalcOptFlow_PipelinePerfTest, TestPerformance)
 
 //------------------------------------------------------------------------------
 
+#ifdef HAVE_OPENCV_VIDEO
+
+PERF_TEST_P_(BackgroundSubtractorPerfTest, TestPerformance)
+{
+    namespace gvideo = cv::gapi::video;
+    initTestDataPath();
+
+    gvideo::BackgroundSubtractorType opType;
+    bool detectShadows = false;
+    double learningRate = -1.;
+    std::string filePath = "";
+    std::size_t testNumFrames = 0;
+    cv::GCompileArgs compileArgs;
+    CompareMats cmpF;
+
+    std::tie(opType, detectShadows, learningRate, filePath, testNumFrames,
+             compileArgs, cmpF) = GetParam();
+
+    const int histLength = 500;
+    double thr = -1;
+    if (opType == gvideo::TYPE_BS_MOG2)
+        thr = 16.;
+    else if (opType == gvideo::TYPE_BS_KNN)
+        thr = 400.;
+    const gvideo::BackgroundSubtractorParams bsp(opType, histLength, thr, detectShadows, learningRate);
+
+    // Video source declaration
+    cv::VideoCapture cap;
+    if (!cap.open(findDataFile(filePath))) throw SkipTestException("Video file can not be opened");
+    cv::Mat frame;
+    cap >> frame;
+
+    // G-API graph declaration
+    cv::GMat in;
+    cv::GMat out = cv::gapi::BackgroundSubtractor(in, bsp);
+    cv::GComputation c(cv::GIn(in), cv::GOut(out));
+
+    // OpenCV Background Subtractor declaration
+    cv::Ptr<cv::BackgroundSubtractor> pOCVBackSub;
+    if (opType == gvideo::TYPE_BS_MOG2)
+        pOCVBackSub = cv::createBackgroundSubtractorMOG2(histLength, thr, detectShadows);
+    else if (opType == gvideo::TYPE_BS_KNN)
+        pOCVBackSub = cv::createBackgroundSubtractorKNN(histLength, thr, detectShadows);
+
+    // Initial step to compile the graph
+    cv::Mat gapiForeground, ocvForeground;
+    c.apply(frame, gapiForeground, std::move(compileArgs));
+    pOCVBackSub->apply(frame, ocvForeground, learningRate);
+    EXPECT_TRUE(cmpF(gapiForeground, ocvForeground));
+    std::size_t frames = 1u;
+
+    // Processing `testNumFrames` amount of frames without measurements:
+    while (frames++ <= testNumFrames && cap.grab())
+    {
+        cap.retrieve(frame);
+        c.apply(frame, gapiForeground);
+        pOCVBackSub->apply(frame, ocvForeground, learningRate);
+        EXPECT_TRUE(cmpF(gapiForeground, ocvForeground));
+    }
```

Comment:
Accuracy checks must be removed from the performance tests. Use accuracy tests for that.

"OCV" code must gone from all perf tests.

Reply:
Yes, a lot of comparison was excess; comparison was removed from the cycle.
However, by design of G-API performance tests, it is necessary to be sure the kernel is working properly while measuring the performance; it's really not my call to change it in this PR; so, I left a single comparison for the validation

Reply:
> Accuracy checks must be removed from the performance tests. Use accuracy tests for that.
> 
> "OCV" code must gone from all perf tests.

@alalek 
The decision to include the verification of an accuracy of  a new GAPI kernels to performance tests was made by the @dmatveev . Please contact with him to discuss.